File: spring-web-java/src/main/java/spring/web/java/dto/response/MemberResponse.java
package spring.web.java.dto.response;


import spring.web.java.constant.Role;
import spring.web.java.domain.Member;

public record MemberResponse(
        Long id,
        String account,
        String name,
        String city,
        String street,
        String zipcode,
        Role role
) {
    public MemberResponse(Member member) {
        this(
                member.getId(),
                member.getAccount(),
                member.getName(),
                member.getAddress().getCity(),
                member.getAddress().getStreet(),
                member.getAddress().getZipcode(),
                member.getRole()
        );
    }
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/dto/response/MemberResponse.kt
package spring.web.kotlin.dto.response

import spring.web.kotlin.constant.Role
import spring.web.kotlin.domain.Member

data class MemberResponse(
    val id: Long,
    val account: String,
    val name: String,
    val city: String,
    val street: String,
    val zipcode: String,
    val role: Role
) {
    constructor(member: Member) : this(
        id = member.id!!,
        account = member.account,
        name = member.name,
        city = member.address.city,
        street = member.address.street,
        zipcode = member.address.zipcode,
        role = member.role
    )
}
